standard/docs/en/schema/merging.md

```diff
@@ -51,106 +52,163 @@ If the record is complete with versioning information, then the versioning secti
+   :jsonpointer: /records/0/versionedRelease/tender/value
+   :expand: value, amount
+   :title: Versioned_values
 
-A reference implementation of the merge routine in Python [is available on GitHub](https://github.com/open-contracting/ocds-merge). 
+```
+
+```eval_rst
+
+.. jsoninclude:: ../examples/merging/versioned.json
+   :jsonpointer:
+   :expand: records, versionedRelease
+   :title: Versioned_release
+
+```
 
-## Versioned data
+The structure of the versioned release is described by the [versioned release schema](../../../../versioned-release-validation-schema.json); note that the `ocid` field's value is not versioned.
 
-There are some situations in which it is important to be able to see how data about a contracting process has changed over time. For example, to identify how contract values have altered, or milestones moved through stages of implementation. 
+### Merge routine
 
-The [versioned release validation schema](../../../../versioned-release-validation-schema.json) provides a model for representing this data.
+1. Get all releases with the same `ocid` value
+1. Order the releases in chronological order by `date`
+1. Create an empty JSON object for the compiled or versioned release
+1. Merge each release (**input**), in order, into the JSON object (**output**), as follows:
 
-In a versioned release, instead of over-writing past values when combining multiple releases, each field (with the exception of the `id` property of objects within an array) becomes an array of objects, indicating the:
+#### Object values
 
-* The date, id and tag of the releases where a field-value pair was first encountered;
-* The value of the field-value pair at that point;
+The release is itself an object, so this case is encountered first.
 
-As a result, the history of any field can be easily read from the data structure.
+If the object is empty in **input**, do nothing. For each field within the object in **input**:
 
-The property `"versionId":true` is used to explicitly declare the cases where an `id` field **should** be versioned (i.e. within an object that is not within an array). 
+* For a compiled release:
+  * If the field in **input** has a value of `null`, remove the field from the object in **output**, if present
+  * If the field isn't in **output**, add the field to the object in **output**, and set it to its value in **input**
+  * If the field is in **output**, merge the field's values in **output** and **input** according to the merge routine
+* For a versioned release:
+  * Merge the field's values in **output** and **input** according to the merge routine; if there is a result, add the field to the object in **output** if not already added, and set it to the result
 
-### Example
+#### Literal values
+
+If the **input** value is neither an object nor an array, then:
+
+* For a compiled release:
+  * If the **input** value is not `null` and is different from the **output** value, replace the **output** value with the **input** value
```

Comment:
> If the **input** value is not `null`...

Since a literal value will never be encountered other than as a property of an object, I think this case is already covered by the "If the field in **input** has a value of `null`..." rule in the object value section.

We should either remove the "If the **input** value is not `null`..." clause here, or if repeating it for clarity, we should also repeat what to do if the field is `null`.

Reply:
Removed in fb0aff6

Reply:
Removed in fb0aff6